test/library/packages/UnitTest/UnitTest.chpl

```diff
@@ -0,0 +1,232 @@
+module UnitTest {
+  use Reflection;
+  /*
+    Assert that a boolean condition is true.  If it is false, prints
+    'assert failed' and halts the program.
+
+    :arg test: the boolean condition
+    :type test: `bool`
+  */
+  pragma "insert line file info"
+  pragma "always propagate line file info"
+  proc assertTrue(test: bool) {
+    if !test then
+      __primitive("chpl_error", c"assert failed");
+  }
+
+  /*
+    Assert that a boolean condition is false.  If it is true, prints
+    'assert failed' and halts the program.
+
+    :arg test: the boolean condition
+    :type test: `bool`
+  */
+  pragma "insert line file info"
+  pragma "always propagate line file info"
+  proc assertFalse(test: bool) {
+    if test then
+      __primitive("chpl_error", c"assert failed");
+  }
+  
+  private
+  proc checkAssertEquality(first, second) {
+    type firstType = first.type;
+    type secondType = second.type;
+
+    if isTupleType(firstType) && isTupleType(secondType) {
+      // both are tuples.
+      assertTupleEqual(first, second);
+      return;
+    }
+    else if isArrayType(firstType) && isArrayType(secondType) {
+      // both are arrays
+      assertArrayEqual(first, second);
+      return;
+    }
+    else if isRangeType(firstType) && isRangeType(secondType) {
+      // both are Range
+      assertRangeEqual(first, second);
+    }
+    else if isString(first) && isString(second) {
+      // both are Strings
+      assertStringEqual(first, second);
+    }
+    else {
+      __baseAssertEqual(first, second);
+    }
+  }
+  
+  private
+  proc all(it : _iteratorRecord) {
```

Comment:
I recommend adding some form of documentation on even private functions, even if just a 1-liner.

Reply:
I have documented them as well. Thanks for the suggestion.